cmd/bugzilla-backporter/main.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/bugzilla"
+	prowConfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/metrics"
+	"k8s.io/test-infra/prow/pjutil"
+)
+
+var (
+	bzbpMetrics = struct {
+		httpRequestDuration *prometheus.HistogramVec
+		httpResponseSize    *prometheus.HistogramVec
+		errorRate           *prometheus.CounterVec
+	}{
+		httpRequestDuration: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Name:    "bugzilla_backporter_http_request_duration_seconds",
+				Help:    "http request duration in seconds",
+				Buckets: []float64{0.0005, 0.001, 0.0025, 0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2},
+			},
+			[]string{"status", "path"},
+		),
+		httpResponseSize: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Name:    "bugzilla_backporter_http_response_size_bytes",
+				Help:    "http response size in bytes",
+				Buckets: []float64{256, 512, 1024, 2048, 4096, 6144, 8192, 10240, 12288, 16384, 24576, 32768, 40960, 49152, 57344, 65536},
+			},
+			[]string{"status", "path"},
+		),
+		errorRate: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "bugzilla_backporter_error_rate",
+				Help: "number of errors, sorted by label/type",
+			},
+			[]string{"error"},
+		),
+	}
+)
+
+func recordError(label string) {
+	labels := prometheus.Labels{"error": label}
+	bzbpMetrics.errorRate.With(labels).Inc()
+}
+
+type options struct {
+	configPath   string
+	registryPath string
+	prowPath     string
+	jobPath      string
+	logLevel     string
+	address      string
+	uiAddress    string
+	gracePeriod  time.Duration
+	cycle        time.Duration
+	validateOnly bool
+	flatRegistry bool
+}
+
+type traceResponseWriter struct {
+	http.ResponseWriter
+	statusCode int
+	size       int
+}
+
+func handleWithMetrics(h http.HandlerFunc) http.HandlerFunc {
```

Comment:
If we now have the identical code in multiple places we should probabl extract it somewhere in a follow-up and re-use it

Reply:
Do you want me to do it as part of this pr itself? If so where should I move that to? Is pkg/api the right place?

Reply:
No, not in this pr, as a follow-up. Maybe a new `pkg/instrumentation` would be a good place